cmd/snap-confine/mount-support-nvidia.c

```diff
@@ -407,13 +408,78 @@ static void sc_mkdir_and_mount_and_bind(const char *rootfs_dir,
 	}
 }
 
+static int sc_mount_nvidia_is_driver_in_dir(const char *dir)
+{
+	char driver_path[512] = { 0 };
+
+	struct sc_nvidia_driver driver;
+
+	// Probe sysfs to get the version of the driver that is currently inserted.
+	sc_probe_nvidia_driver(&driver);
+
+	// If there's no driver then we should not bother ourselves with finding the
+	// matching library
+	if (driver.major_version == 0) {
+		return 0;
+	}
+	// Probe if a well known library is found in directory dir
+	sc_must_snprintf(driver_path, sizeof driver_path,
+			 "%s/" LIBNVIDIA_GLCORE_SO_PATTERN, dir,
+			 driver.major_version, driver.minor_version);
+
+	if (access(driver_path, F_OK) == 0) {
+		debug("nvidia library detected at path %s", driver_path);
+		return 1;
+	}
+	return 0;
+}
+
 static void sc_mount_nvidia_driver_multiarch(const char *rootfs_dir)
 {
-	// Attempt mount of both the native and 32-bit variants of the driver if they exist
-	sc_mkdir_and_mount_and_bind(rootfs_dir, "/usr/lib/nvidia",
-				    SC_LIBGL_DIR);
-	sc_mkdir_and_mount_and_bind(rootfs_dir, "/usr/lib32/nvidia",
-				    SC_LIBGL32_DIR);
+	const char *native_libdir = NATIVE_LIBDIR "/" HOST_ARCH_TRIPLET;
+	const char *lib32_libdir = NATIVE_LIBDIR "/" HOST_ARCH32_TRIPLET;
+
+	if ((strlen(HOST_ARCH_TRIPLET) > 0) &&
```

Comment:
I didn't notice this before, why are we testing for the length of a compile-time fixed constant?

Reply:
Because I didn't want to introduce #ifdefs and end up with awkwardly laid out code :/ Since HOST_ARCH_TRIPLET is a define for `""` in the not-enabled case, a half-decent compiler will be able to drop the branch entirely.